app/src/main/java/com/github/libretube/ui/dialogs/UpdateAvailableDialog.kt

```diff
@@ -1,33 +1,35 @@
 package com.github.libretube.ui.dialogs
 
 import android.app.Dialog
+import android.content.Context
+import android.content.DialogInterface
 import android.content.Intent
 import android.os.Bundle
+import androidx.core.content.ContextCompat.startActivity
 import androidx.core.net.toUri
 import androidx.fragment.app.DialogFragment
 import com.github.libretube.R
-import com.github.libretube.constants.IntentData
-import com.github.libretube.extensions.parcelable
-import com.github.libretube.obj.update.UpdateInfo
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 
 class UpdateAvailableDialog : DialogFragment() {
-    private lateinit var updateInfo: UpdateInfo
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        updateInfo = requireArguments().parcelable(IntentData.updateInfo)!!
-    }
-
-    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-        return MaterialAlertDialogBuilder(requireContext())
-            .setTitle(context?.getString(R.string.update_available, updateInfo.name))
-            .setMessage(context?.getString(R.string.update_available_text))
-            .setNegativeButton(R.string.cancel, null)
-            .setPositiveButton(context?.getString(R.string.okay)) { _, _ ->
-                val intent = Intent(Intent.ACTION_VIEW).setData(updateInfo.htmlUrl.toUri())
-                startActivity(intent)
-            }
+    fun onCreateDialog(
+        changelog: String,
+        releaseUrl: String,
+        context: Context,
+    ): Dialog {
```

Comment:
Please refactor this so that the arguments are passed as intent extras, basically as it's been before.

This implementation might crash/dismiss the dialog once the theme is changed on certain devices.

Reply:
i have used the intent bundle for passing data and have tested well on android 9 and 13 , the lint is asking for SNAKE-CASE  in intent data for these two things(changelog and url), should i do it or ignore it because all others are in camelCase